server/routerlicious/packages/lambdas/src/deli/lambda.ts

```diff
@@ -1209,10 +1210,23 @@ export class DeliLambda extends TypedEventEmitter<IDeliLambdaEvents> implements
         } as any;
         if (message.operation.type === MessageType.Summarize || message.operation.type === MessageType.NoClient) {
             const augmentedOutputMessage = outputMessage as ISequencedDocumentAugmentedMessage;
-            if (message.operation.type === MessageType.Summarize ||
-                this.serviceConfiguration.scribe.generateServiceSummary) {
-                // only add additional content if scribe will use this op for generating a summary
-                // NoClient ops are ignored by scribe when generateServiceSummary is disabled
+
+            // only add additional content if scribe will use this op for generating a summary
+            // NoClient ops are ignored by scribe when generateServiceSummary is disabled
+            let addAdditionalContent = false;
+
+            if (this.serviceConfiguration.scribe.generateServiceSummary) {
+                addAdditionalContent = true;
+            } else if (message.operation.type === MessageType.Summarize) {
+                // no need to add additionalContent for summarize messages using the single commit flow
+                // because scribe will not be involved
+                if (!this.serviceConfiguration.deli.skipSummarizeAugmentationForSingleCommmit ||
+                    !(JSON.parse(message.operation.contents) as ISummaryContent).details?.includesProtocolTree) {
```

Comment:
Client can crash lamba by pushing invalid content here, right? Should it be resilient against it?